test/pipelines/unit/test_validation_pipeline_io.py

```diff
@@ -177,6 +177,20 @@ def test_validate_pipeline_input_only_expected_input_is_present():
     with pytest.raises(ValueError, match="The input value of comp2 is already sent by node comp1"):
         pipe.run({"comp1": {"value": 1}, "comp2": {"value": 2}})
 
+def test_validate_pipeline_input_only_expected_input_is_present_falsy():
+    pipe = Pipeline()
+    pipe.add_component("comp1", Double())
+    pipe.add_component("comp2", Double())
+    pipe.connect("comp1", "comp2")
+    with pytest.raises(ValueError, match="The input value of comp2 is already sent by node comp1"):
+        pipe.run({"comp1": {"value": 1}, "comp2": {"value": 0}})
+
+def test_validate_pipeline_falsy_input():
+    pipe = Pipeline()
+    pipe.add_component("comp", Double())
+    pipe.run({"comp": {"value": 0}})
+    with pytest.raises(ValueError, match="Missing input: comp.value"):
+        pipe.run({"comp": {}})
```

Comment:
Sorry I'm being a bit pedantic here :sweat_smile:  Let's split these two tests

```suggestion
def test_validate_pipeline_falsy_input_present():
    pipe = Pipeline()
    pipe.add_component("comp", Double())
    assert pipe.run({"comp": {"value": 0}}) == {"comp": {"value": 0}}
        
def test_validate_pipeline_falsy_input_missing():
    pipe = Pipeline()
    pipe.add_component("comp", Double())
    with pytest.raises(ValueError, match="Missing input: comp.value"):
        pipe.run({"comp": {}})
```